lib/MaxMind/DB/Writer/Tree.pm

```diff
@@ -287,12 +331,12 @@ sub new_from_frozen_tree {
 
     $params->{database_type} = $database_type if defined $database_type;
     $params->{description}   = $description   if defined $description;
+    $params->{merge_strategy} = $merge_strategy if defined $merge_strategy;
 
     my $tree = _thaw_tree(
         $filename,
         $params_size + 4,
-        map { $params->{$_} }
-            qw( ip_version record_size merge_record_collisions ),
+        map { $params->{$_} } qw( ip_version record_size merge_strategy ),
```

Comment:
I am pretty sure we are the only users of the frozen trees. I guess we could check if `merge_record_collisions` exists and, if so, set `merge_strategy` to `toplevel` if it is not explicitly set  though.